lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -2240,21 +2254,52 @@ deletePendingOrExpiredTx wid tid = do
         Nothing -> fromIntegral <$> deleteWhereCount
             ((TxMetaStatus <-. [W.Pending, W.Expired]):filt)
 
+listPendingLocalTxSubmissionQuery
+    :: W.WalletId
+    -> SqlPersistT IO [(W.SlotNo, LocalTxSubmission)]
+listPendingLocalTxSubmissionQuery wid = fmap unRaw <$> rawSql query params
+  where
+    query = "SELECT meta.slot,?? FROM (select tx_id, status, slot, MAX(slot) AS latest_slot FROM tx_meta WHERE wallet_id=? GROUP by tx_id) AS meta INNER JOIN local_tx_submission ON meta.tx_id=local_tx_submission.tx_id WHERE status=? AND slot=latest_slot"
+    params = [toPersistValue wid, toPersistValue W.Pending]
+    unRaw (Single sl, Entity _ tx) = (sl, tx)
```

Comment:
I'd love a comment which clarifies what is expected from this function. Looking at the query, I assume it's meant to return all most recent submissions for pending transactions (which is sort of hinted by the name already, but some clarifications would be much appreciated :pray: )

I actually went to see that function back again because I was getting a bit confused by the logic on the wallet side, thinking that we might be re-submitting transactions that are _in-ledger_ (which isn't a major issue per se, but is however useless). Now I see that this actually filters transactions by status, only considering _pending_ ones. Seems good.

Reply:
I'll add a comment to the query - it definitely needs explanation.